### PR TITLE
helm: Support dynamic AWS database discovery in teleport-kube-agent chart

### PR DESCRIPTION
**Pull request by mantoine96**
Hi,

I've been using the `teleport-kube-agent` chart to deploy a pool of agents to proxy connections to DBs. 

Now that automated discovery of AWS databases is available in Teleport 8.0, I wanted to leverage it from the `teleport-kube-agent` chart, which doesn't currently support it. 

Therefore this PR adds a new value in `teleport-kube-agent`, `awsDatabases` (I welcome feedback on the naming of this value).

I've also added the file for the linter, as well as updated the documentation to reflect the new value.

Finally, one last note:
`semverCompare` was breaking on the `appVersion: 8.0.0-alpha.1`. I've bumped it to `8.1.0` since that is the latest release for Teleport (and since that is the only way to actually exercise the new templating without removing `semverCompare`) , but I can 100% remove that if not desired.

### CODE REVIEW

File: docs/pages/kubernetes-access/helm/reference.mdx [hunk 1]
```diff
@@ -1480,6 +1480,64 @@ You can specify multiple apps by adding additional list elements.
...
+
+`awsDatabases` is a YAML list object detailing the filters for the AWS databases that should be discovered and proxied by Teleport Database access.
+
+You can specify multiple database filters by adding additional list elements.
+
+<Tabs>
+  <TabItem label="values.yaml">
+  ```yaml
+  awsDatabases:
+    - types: ["rds", "aurora"]
```
Comment (greedy52) [inline docs/pages/kubernetes-access/helm/reference.mdx:1497]: currently the `rds` type will discover both RDS dbs and Aurora clusters (no type `aurora`). `redshift` will likely come in 9.0.

https://github.com/gravitational/teleport/blob/538fcaa98007266e7940df51ca6a2e584cf86be1/lib/services/matchers.go#L60-L65

Comment (thehunt33r) [reply docs/pages/kubernetes-access/helm/reference.mdx]: Fixed, thanks!

File: examples/chart/teleport-kube-agent/Chart.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 name: teleport-kube-agent
 apiVersion: v2
-version: "8.0.0-alpha.1"
-appVersion: "8.0.0-alpha.1"
+version: "8.1.0"
+appVersion: "8.1.0"
```
Comment (zmb3) [inline examples/chart/teleport-kube-agent/Chart.yaml:4]: @webvictim do we update this as part of a release? If now how do we keep the chart version in sync with the Teleport version?

File: examples/chart/teleport-kube-agent/templates/config.yaml [hunk 3]
```diff
@@ -53,6 +53,23 @@ data:
     db_service:
       {{- if contains "db" (.Values.roles | toString) }}
       enabled: true
+      {{- if (or .Values.awsDatabases .Values.databases) }}
```
Comment (zmb3) [inline examples/chart/teleport-kube-agent/templates/config.yaml:56]: Why the check for `.Values.databases` here?

File: examples/chart/teleport-kube-agent/Chart.yaml (hunk 2 above)
Comment (webvictim) [reply examples/chart/teleport-kube-agent/Chart.yaml:4]: The Helm chart version gets updated as part of the `version.mk` Makefile: https://github.com/gravitational/teleport/blob/master/version.mk#L33-L45

Comment (thehunt33r) [reply examples/chart/teleport-kube-agent/Chart.yaml:4]: Ah that explains why `8.0.0-alpha.1` was still there and not `8.1.0` or `8.1.1-beta.1` as I was expecting. `master` still has `8.0.0-alpha.1` as its chart version, and that's what I started from.

I will undo this change then 👍

**Pull request closed without merging**